packages/sdk/client/src/packlets/client/iframe-service-proxy.ts

```diff
@@ -156,12 +165,32 @@ export class IFrameClientServicesProxy implements ClientServicesProvider {
     );
 
     if (!this._iframe) {
-      const iframeId = `__DXOS_CLIENT_${PublicKey.random().toHex()}__`;
-      this._iframe = createIFrame(source.toString(), iframeId, { allow: 'clipboard-read; clipboard-write' });
       const res = await fetch(source);
       if (res.status >= 400) {
-        throw new RemoteServiceConnectionTimeout();
+        throw new RemoteServiceConnectionError(`Failed to fetch ${source}`, { source, status: res.status });
       }
+      const iframeId = `__DXOS_CLIENT_${PublicKey.random().toHex()}__`;
+      this._iframe = createIFrame(source.toString(), iframeId, { allow: 'clipboard-read; clipboard-write' });
```

Comment:
is there any chance that the postMessage might happen between this line and the handler being added